eodag/rest/utils.py

```diff
@@ -1046,6 +1050,126 @@ def fetch_collection_queryable_properties(
+    :rtype dict
+    """
+    search_plugin = next(
+        eodag_api._plugins_manager.get_search_plugins(provider=provider)
+    )
+    search_type = search_plugin.config.type
+    if search_type == "StacSearch":
+        queryables_url = search_plugin.config.api_endpoint.replace(
+            "/search", "/queryables"
+        )
+    else:
+        queryables_url = getattr(search_plugin.config, "queryables_endpoint", "")
+    if not queryables_url:
+        logger.warning("no url for queryables found for provider %s", provider)
+        return queryables
+    try:
+        if hasattr(search_plugin, "auth"):
+            headers = getattr(search_plugin.auth, "headers", {})
+            headers.update(USER_AGENT)
+        else:
+            headers = USER_AGENT
+        res = requests.get(queryables_url, headers=headers)
+        res.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        if err.response.status_code == 404:
+            logger.info("provider %s does not support queryables", provider)
+            return queryables
+        else:
+            raise RequestError(str(err))
+    else:
+        provider_queryables = res.json()["properties"]
+        return _format_provider_queryables(provider_queryables, queryables)
+
+
+def add_provider_product_type_queryables(
+    provider: str, product_type: str, queryables: dict
+):
+    """Add the queryables fetched from the given provider for the given product type
+    to the default queryables derived from the metadata mapping.
+    If the queryables endpoint is not supported by the provider, the original queryables are returned.
+    :param provider: The provider.
+    :type provider: str
+    :param product_type: EODAG product type
+    :type product_type: str
+    :param queryables: default queryables to which provider queryables will be added
+    :type queryables: dict
+    :returns queryable_properties: A dict containing the formatted queryable properties
+                                   including queryables fetched from the provider.
+    :rtype dict
+    """
+    search_plugin = next(
+        eodag_api._plugins_manager.get_search_plugins(provider=provider)
+    )
+    search_type = search_plugin.config.type
+    provider_product_type = search_plugin.config.products.get(product_type, {}).get(
+        "productType", None
+    )
+    if not provider_product_type:
+        logger.warning(
+            "provider product type mapping for product type %s not found", product_type
+        )
+        provider_product_type = product_type
+    if search_type == "StacSearch":
+        api_url = search_plugin.config.api_endpoint.replace("/search", "/")
+        queryables_url = (
+            api_url + "collections/" + provider_product_type + "/queryables"
+        )
+    else:
+        queryables_url = getattr(
+            search_plugin.config, "queryables_endpoint", ""
+        ).format(product_type=provider_product_type)
```

Comment:
do not handle queryables for non-STAC providers

Reply:
changed